### DISCUSSION TITLE
Orientation, rotation by 180°

### DISCUSSION THREAD

**Discussion by matrais11** [olikraus/u8g2#1934] [Ideas]
Hi olikraus,

I really appreciate your work and intensity of replying all questions here. Keep up the good work, it helped me a lot, although I'm a beginner :)

I bought [this display](https://www.audiophonics.fr/en/screens-vu-meters/white-spi-iic-256x64-oled-312-screen-p-14604.html) last year and managed it to work in the fast mode via SPI and 6 wires. Uploaded a demo and it works flawlessly ([youtube](https://www.youtube.com/watch?v=VX7gFL9v1qo
)). 

Now, I have an issue: I had my amplifier chassis CNC'd, all slots and holes are cut into the chassis and I forgot to count with the ports of the display being on the left side of the PCB. And it's just blocking the push-button (or vice-versa) so I need to physically rotate the display to fit into the same panel cutout while connectors are on the right.

This means: the display would need to be rotated by 180° CW/CCW (same result of course). 

Is there a way, a simple method of defining the rotation/direction of the whole display, so that my graphics and text are not upside down ? It's not a "horizontal flip" like in Photoshop (we couldn't read text then, all mirrored), but a full rotation by 180˚. 

Or can I already somehow reconfigure it in u8g2.h or elsewhere maybe ? (I'm a beginner, just sayin', but I try to deal with instructions).

**Comment by matrais11** [comment]
Uh it's U8G2_R**2** in 
U8G2_SSD1322_NHD_256X64_F_4W_HW_SPI u8g2(**U8G2_R2**, /* cs=*/ 15, /* dc=*/ 4, /* reset=*/ 5);  // Enable U8G2_16BIT in u8g2.h

Sorry. Answered.

**Comment by olikraus** [comment]
Additionally, there is also setFlipMode: https://github.com/olikraus/u8g2/wiki/u8g2reference#setflipmode
